### DISCUSSION TITLE
Ender3 S1 Plus support and psu control

### DISCUSSION THREAD

**Discussion by pico45** [mriscoc/Ender3V2S1#765] [Q&A]
I have an ender3 s1 plus with v2 display and played the professional firmware on it. but i can't change the bed size to 300x300x300 in physical settings. is there a customized firmware for the Ender3 S1 plus???
my second question: is it possible to activate a ps_pin (psu control) (PA13 or PA14) so that the Ender 3 turns off after printing ???

**Comment by gsuresh2u** [comment]
For psu control, you need to compile your own firmware
no precompiled firmware will work,
you need to enable #define PSU_CONTROL
after enabling it, you can enable  other like #define AUTO_POWER_CONTROL, etc

**Comment by TheWebMachine** [comment]
This firmware works fine on my S1 Plus. In order to increase bed size, you need to first properly set the Min/Max for all axes. Then you can set proper bed size after that. I use the following Min/Max settings on my plus:

X-min -8
Y-min -3
X-max 323
Y-max 310
Z-max 310

Also don't forget to change Mesh Leveling Insets after you've set proper Min/Max and Bed size.

So, in summary, setup physical params in the following order: Min/Max, Physical Bed Size, Mesh Insets

**Reply by pico45** [reply]
Thank you very much for your answers. especially TheWebMachine. after changing the stats, I was now able to change the physical stats as well. Excellent.
now I only have the problem with the PS_Pin. does anyone have the source code of the current firmware Ender3S1-F4-20221222.bin so that i can activate the ps_pin?? a question for mriscoc. Is it possible to already activate the ps_pin for the Ender3 S1 on PA13 or PA14 with new updates. or to be able to adjust via the display. then everyone could decide for themselves whether to use the pin or not.

**Reply by TheWebMachine** [reply]
Mriscoc can elaborate when he has a moment, but you'll find that the F4 boards have very little flash space available and it's difficult to add anything that needs additional menus. He has added as many features as he can for F4 boards, but had to preclude a lot as well.

You'll want to check out his special configurations repo for some insight on building your own.
https://github.com/mriscoc/Special_Configurations

**Reply by mriscoc** [reply]
> anyone have the source code of the current firmware

Source code is in the main repository https://github.com/mriscoc/Ender3V2S1 and configuration files in https://github.com/mriscoc/Special_Configurations

**Reply by pico45** [reply]
Many, many thanks TheWebMachine and MRISCOC for your help. could solve the problems.
